### ISSUE TITLE
Empty BAM file works with genomecov from 2.27.1, but not 2.28.0

### ISSUE THREAD

**Issue by rpetit3** [arq5x/bedtools2#718]
Hello!

I used conda to install a program called [IS_Mapper](https://github.com/jhawkey/IS_mapper/). In it `genomecov` is used, and depending on the version of bedtools used, the input BAM files may or maynot work.

In v2.28.0 (conda version - 2.28.0-hdf88d34_0) I get the following error:

```
bedtools genomecov -ibam MRSABB_00149.28_left_MRSABB_00149.28_1.sorted.bam -bg
Failed to open BAM file IS256/MRSABB_00149.28_left_MRSABB_00149.28_1.sorted.bam
```

With  v2.27.1 (conda version - 2.27.1-he860b03_3), `genomecov` works and I get:
```
bedtools genomecov -ibam MRSABB_00149.28_left_MRSABB_00149.28_1.sorted.bam
MRSABB_00149.28_1       0       284157  284157  1
genome  0       284157  284157  1
```

Here's the `samtools flagstat` (its an empty BAM file)
```
samtools flagstat MRSABB_00149.28_left_MRSABB_00149.28_1.sorted.bam
0 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
0 + 0 mapped (N/A : N/A)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (N/A : N/A)
0 + 0 with itself and mate mapped
0 + 0 singletons (N/A : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
```

Switching back to 2.27.1 is not a problem for me, but thought it might be useful to report this. Let me know if you need the BAM file, I can share it.

Thank you!
Robert

**Comment by arq5x** [comment]
Thanks for reporting, I think this is the result of the new BAM interface we wrote on top of htslib. Will dig in soon.

**Comment by arq5x** [comment]
@38 would you mind looking into this?  I think it has to do with a failure at BAM open when BAM is empty.

**Comment by arq5x** [comment]
This should be fixed in master via #725 from @38. Please test on your end and reopen if you still have a problem.
